packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts

```
@@ -0,0 +1,35 @@
+import * as A from './actions'
+import { put, select } from 'redux-saga/effects'
+import { selectors } from 'data'
+
+import { initialize } from 'redux-form'
+import { nth } from 'ramda'
+
+export default () => {
+  const initializeInterest = function * ({
+    payload
+  }: ReturnType<typeof A.initializeInterest>) {
+    let defaultAccountR
+
+    switch (payload.coin) {
+      case 'BTC':
+        const accountsR = yield select(
+          selectors.core.common.btc.getAccountsBalances
+        )
+        const defaultIndex = yield select(
+          selectors.core.wallet.getDefaultAccountIndex
+        )
+        defaultAccountR = accountsR.map(nth(defaultIndex))
+        break
+    }
```

Might want to throw an error for other coins if their not set up yet.